### DISCUSSION TITLE
Purpose of cilium_net@cilium_host veth pair

### DISCUSSION THREAD

**Discussion by ttc0419** [cilium/cilium#36069] [Q&A] [answered]
Are there any documentation about what's the cilium_net@cilium_host veth pair for? They are both in the host namespace, why? If it's just a dummy interface why not just use a bridge etc?

**Comment by joestringer** [comment, marked as answer]
This device pair provides a target location for routing traffic from the host towards the Cilium-managed network. Depending on a few settings like IPAM and "endpoint routes" mode, it is common to route entire IP prefixes towards this device and then there is one common place that Cilium can apply eBPF programs to intercept traffic and implement networking functionality. The devices are also used when traffic ingressing a Pod requires layer 7 interception, as Cilium needs to redirect the traffic away from the Pod towards the host in order to deliver the traffic to the proxy. There may be other use cases as well.

I don't think there's a specific reason not to use a bridge device, other than that Cilium doesn't need a bridge. Cilium just needs a device that the Linux stack can route towards, and similarly a device where the Linux stack can receive traffic from Cilium. Another option might be a dummy device, but I vaguely recall there are some limitations in the Linux implementation of dummy devices that makes this not an option.
